### PR TITLE
feat: check for data type in transformers

### PR DESCRIPTION
**Pull request by ilhaoni**

### CODE REVIEW

**Comment by ddwsd** [comment]
Remember to add tests for path transformer too!

File: src/functionHandlers/proxy/transformers/bodyTransformer.test.ts [hunk 1]
```diff
@@ -50,6 +51,35 @@ describe("bodyTransformer with hashedMapper", () => {
       ]
     });
   });
+
+  it("leave id as it is and transforms requests with id if validation type is ffth-id", async () => {
```
Comment (ddwsd) [inline src/functionHandlers/proxy/transformers/bodyTransformer.test.ts:55]: nit – use plural case for "leave" here haha

File: src/functionHandlers/proxy/transformers/bodyTransformer.test.ts [hunk 2]
```diff
@@ -50,6 +51,35 @@ describe("bodyTransformer with hashedMapper", () => {
...
+      foo: "bar"
+    });
+    expect(await hashedMapper(input)).toEqual({
+      id: "HOW_NOW_BROWN_COW_abc123",
+      foo: "bar",
+      hashedOriginalId: "abc123"
+    });
+  });
+
+  it("leave id as it is and transforms requests with ids if validation type is ffth-id", async () => {
```
Comment (ddwsd) [inline src/functionHandlers/proxy/transformers/bodyTransformer.test.ts:69]: use plural case here too!

File: src/functionHandlers/proxy/transformers/bodyTransformer.ts [hunk 3]
```diff
@@ -1,21 +1,35 @@
...
 
-    body.id = await transformer(hashedId);
-    body.hashedOriginalId = hashedId;
-  }
-  if (ids) {
-    const hashedIds = ids.map(identifierStringToSaltedHash);
+  if (config.appParameters.validationType === "FFTH-ID") {
+    if (id) {
+      body.id = await transformer(id);
+      body.hashedOriginalId = id;
```
Comment (ddwsd) [inline src/functionHandlers/proxy/transformers/bodyTransformer.ts:12]: this means that the qr code would be the hashedoriginalid in storage table right haha

File: src/functionHandlers/proxy/transformers/bodyTransformer.test.ts (hunk 1 above)
Comment (ilhaoni) [reply src/functionHandlers/proxy/transformers/bodyTransformer.test.ts:55]: ill change the one with multiple ids :P

Comment (ilhaoni) [reply src/functionHandlers/proxy/transformers/bodyTransformer.test.ts:55]: but this is a single id HAHA

File: src/functionHandlers/proxy/transformers/bodyTransformer.ts (hunk 3 above)
Comment (ilhaoni) [reply src/functionHandlers/proxy/transformers/bodyTransformer.ts:12]: yup!

File: src/functionHandlers/proxy/transformers/bodyTransformer.test.ts (hunk 1 above)
Comment (ddwsd) [reply src/functionHandlers/proxy/transformers/bodyTransformer.test.ts:55]: i think the "it" is referring to the code, not the id 😆

**Pull request merged**